File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        first = strs[0]
        for i in range(len(first)):
            c = first[i]
            for s in strs[1:]:
                if i == len(s) or s[i] != c:
                    return first[:i]
        return first
                

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string encoded = "" ;
        for(const string& s: strs){
            encoded += to_string(s.size())+"#" + s;
        }
        return encoded;
    }

    vector<string> decode(string s) {
        vector<string> decoded;
        int i =0;
        while(i<s.size()){
            int j =i;
            while(s[j] != '#'){
                j++;
            }
            //extract length of next string
            int length = stoi(s.substr(i,j-i));

            i=j+1;
            
            decoded.push_back(s.substr(i,length));

            i+= length;
        }
        return decoded;
    }
};
